src/production/logic.py

```diff
@@ -173,23 +173,46 @@ def replace_galley_file(article, request, galley, uploaded_file):
         messages.add_message(request, messages.WARNING, 'No file was selected.')
 
 
-def save_galley_image(galley, request, uploaded_file, label="Galley Image", fixed=False):
+def save_galley_image(
+        galley,
+        request,
+        uploaded_file,
+        label="Galley Image",
+        fixed=False,
+        check_for_existing_images=False
+):
+    filename = uploaded_file.name
+    # Check if an image with this name already exists for this galley.
+
+    if check_for_existing_images and galley.images.filter(
+        original_filename=filename,
+    ).exists():
+        messages.add_message(
+            request,
+            messages.WARNING,
+            f'An image called {filename} already exists. Use the replace '
+            f'file function to upload a new version.',
+        )
+        return
 
     if fixed:
-        filename = request.POST.get('file_name')
         uploaded_file_mime = files.check_in_memory_mime(uploaded_file)
         expected_mime = files.guess_mime(filename)
 
         if not uploaded_file_mime == expected_mime:
-            messages.add_message(request, messages.WARNING, 'The file you uploaded does not match the mime of the '
-                                                            'file expected.')
+            messages.add_message(
+                request,
+                messages.WARNING,
+                'The file you uploaded does not match the mime of the '
+                'file expected.',
```

Comment:
This is a good opportunity to make this text a little more user-friendly. Most people don't know what MIME types are, and we also aren't showing users potentially helpful information (e.g. what the expected type is). Maybe we could do something like `f'The file you uploaded does not have the expected type: { expected_mime }.'`

Reply:
Thanks Joe. Will update.